pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -159,7 +159,7 @@ func ApplyPipelineTaskContexts(pt *v1.PipelineTask) *v1.PipelineTask {
 	}
 	pt.Params = pt.Params.ReplaceVariables(replacements, map[string][]string{}, map[string]map[string]string{})
 	if pt.IsMatrixed() {
-		pt.Matrix.Params = pt.Params.ReplaceVariables(replacements, map[string][]string{}, map[string]map[string]string{})
+		pt.Matrix.Params = pt.Matrix.Params.ReplaceVariables(replacements, map[string][]string{}, map[string]map[string]string{})
```

Comment:
do we have unit tests in apply_test.go to cover this?

Reply:
The existing unit tests does cover this.

![Screenshot 2023-09-01 at 12 52 56 PM](https://github.com/tektoncd/pipeline/assets/206285/1e1cfa75-e268-49c7-8d42-7e5f3d78967c)

But the existing test was not able to catch this issue. This bug surfaces with when a non-matrix parameter is added to the list of params belonging to a matrix. We have added a [test](https://github.com/tektoncd/pipeline/pull/7064/files#diff-a64a19b939f791920d235cbf5bf2e4905d4814af06f21401b27af11351afbd0bR8285) to catch this error (`parameters of type array only are allowed, but param version has type string`) and address it as part of this PR.  

If we want to still would like to add a test locally, we could add the following test case. I think this is not necessary. Let me know you think.

```
	description: "params and matrix params are not overwritten",
		pt: v1.PipelineTask{
			Params: v1.Params{{
				Name:  "params-1",
				Value: *v1.NewStructuredValues("0"),
			}},
			Matrix: &v1.Matrix{
				Params: v1.Params{{
					Name:  "params-2",
					Value: *v1.NewStructuredValues("5"),
				}},
			},
		},
		want: v1.PipelineTask{
			Params: v1.Params{{
				Name:  "params-1",
				Value: *v1.NewStructuredValues("0"),
			}},
			Matrix: &v1.Matrix{
				Params: v1.Params{{
					Name:  "params-2",
					Value: *v1.NewStructuredValues("5"),
				}},
			},
```